### PR TITLE
Add trimal

### PR DESCRIPTION
**Pull request by B0r1sD**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

This is a collaboration together with @pauldg and @nicodr97. @pauldg made the original wrapper, that resides at https://github.com/usegalaxy-be/galaxytools/tree/main/trimal.

What I changed so far:
- Added version macros
- reStructured text as is was giving warnings when using a bullet list (which looks way nicer for options...) due to a dot being present in some options (e.g. `phylip3.2_m10`) so changed to a simpler [definition list](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#definition-lists). I didn't manage to escape or cheese the strict formatting rules.
- Ran the tests, they are successful.
- Updated profile to 23.00. Think it's best practice to have this Galaxy version as high as possible?
- Changed citation to simpler DOI style.

`Planemo lint` now only fails on '\<when /> block' not being present at the 'Conditional [trimming_mode]'. Could you add them @nicodr97 or @pauldg as was done at the 'custom' value?  

WIP will be removed after these changes and [IUC's checklist](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/integration_checklist.html) have been addressed.

Would also like to point out that Trimal 1.5.0 exists, but think we should get 1.4.1 wrapper properly finished and then release a new updated wrapper for the 1.5.0 version.

### CODE REVIEW

File: tools/trimal/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+categories:
+- Phylogenetics
+description: Tool for automated alignment trimming
+homepage_url: http://trimal.cgenomics.org/trimal
+long_description: trimAl is a tool for the automated removal of spurious sequences
+  or poorly aligned regions from a multiple sequence alignment.
+name: trimal
+owner: padge
```
Comment (bgruening) [inline tools/trimal/.shed.yml:8]: iuc?

File: tools/trimal/trimal.xml [hunk 2]
```diff
@@ -0,0 +1,159 @@
+<tool id="trimal" name="trimAl" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.00" python_template_version="3.5">
```
Comment (bgruening) [inline tools/trimal/trimal.xml:1]:
```suggestion
<tool id="trimal" name="trimAl" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
```

File: tools/trimal/trimal.xml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    <description>A tool for automated alignment trimming</description>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:2]:
```suggestion
    <description>for automated alignment trimming</description>
```

File: tools/trimal/trimal.xml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    <macros>
+        <token name="@TOOL_VERSION@">1.4.1</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">trimal</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[    
+        trimal -in $input1 -out $trimmed_output -htmlout $html_summary ${out_format_selector}
```
Comment (bgruening) [inline tools/trimal/trimal.xml:11]:
```suggestion
        trimal -in '$in' -out '$trimmed_output' -htmlout '$html_summary' ${out_format_selector}
```

File: tools/trimal/trimal.xml [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            -simthreshold $trimming_mode.simthreshold
+            -cons $trimming_mode.cons
+        #else:
+            $trimming_mode.mode_selector
+        #end if
+        ;
+
+    ]]></command>
+    <inputs>
+        <param name="input1" type="data" format="fasta,clustal,pir,phylip,nexus,mega" multiple="false" label="Alignment file (clustal, fasta, NBRF/PIR, nexus, phylip3.2, phylip)" optional="false" />
```
Comment (bgruening) [inline tools/trimal/trimal.xml:23]:
```suggestion
        <param argument="-in" type="data" format="fasta,clustal,pir,phylip,nexus,mega" label="Alignment file (clustal, fasta, NBRF/PIR, nexus, phylip3.2, phylip)" />
```

File: tools/trimal/trimal.xml [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+                <option value="-strictplus">strictplus. (see User Guide)</option>
+                <option value="-automated1">automated</option>
+                <option value="custom">Custom mode</option>
+            </param>
+            <when value="custom">
+                <param argument="-gapthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="gap threshold" help="1 - (fraction of sequences with a gap allowed). Range: [0 - 1]" multiple="false"/>
+                <param argument="-simthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="similarity threshold" help="Minimum average similarity allowed. Range: [0 - 1]" multiple="false"/>
+                <param argument="-cons" type="integer" optional="true" value="50" min="0" max="100" label="min conservance percentage" help="Minimum percentage of the positions in the original alignment to conserve." multiple="false"/>
+                
+            </when>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:39]:
```suggestion
            </when>
            <when value="-nogaps"/>
```
and the others need to be added

File: tools/trimal/trimal.xml [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+                <when input="out_format_selector" value="-clustal" format="clustal" />
+                <when input="out_format_selector" value="-mega" format="mega" />
+                <when input="out_format_selector" value="-nbrf" format="pir" />
+                <when input="out_format_selector" value="-nexus" format="nexus" />
+            </change_format>
+        </data>
+        <data name="html_summary" format="html" label="trimal html summary."/>
+    </outputs>
+    <tests>
+        <test>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:76]: please add expected_num_outputs

File: tools/trimal/trimal.xml [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+                <when input="out_format_selector" value="-fasta" format="fasta" />
+                <when input="out_format_selector" value="-fasta_m10" format="fasta" />
+                <when input="out_format_selector" value="-phylip" format="phylip" />
+                <when input="out_format_selector" value="-phylip_m10" format="phylip" />
+                <when input="out_format_selector" value="-phylip_paml" format="phylip" />
+                <when input="out_format_selector" value="-phylip_paml_m10" format="phylip" />
+                <when input="out_format_selector" value="-phylip3.2" format="phylip" />
+                <when input="out_format_selector" value="-phylip3.2_m10" format="phylip" />
+                <when input="out_format_selector" value="-clustal" format="clustal" />
+                <when input="out_format_selector" value="-mega" format="mega" />
```
Comment (bgruening) [inline tools/trimal/trimal.xml:68]: does the mega format exists in Galaxy?

Comment (bgruening) [reply tools/trimal/trimal.xml:68]: and the othes?

File: tools/trimal/trimal.xml [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+            <param name="gapthreshold" value="0.5" />
+            <param name="simthreshold" value="0.5" />
+            <param name="cons" value="5" />
+            <param name="out_format_selector" value="-phylip_paml_m10" />
+            <output name="trimmed_output" file="custom_trimmed_example.009.AA.phy" />
+            <output name="html_summary" file="custom_trimmed_example.009.AA.html"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+Modes
```
Comment (bgruening) [inline tools/trimal/trimal.xml:95]: This help is not very useful for users. The parameters are already described. Maybe some text what this tool is doing etc ...

File: tools/trimal/trimal.xml (hunk 8 above)
Comment (nicodr97) [reply tools/trimal/trimal.xml:68]: Some of them don't exist in Galaxy. Should we just remove them from here?

File: tools/trimal/.shed.yml (hunk 1 above)
Comment (B0r1sD) [reply tools/trimal/.shed.yml:8]: @pauldg do we have your blessing for this?

**Comment by bgruening** [comment]
If you think its ready, please take it out of draft and remove the WIP

File: tools/trimal/trimal.xml (hunk 8 above)
Comment (B0r1sD) [reply tools/trimal/trimal.xml:73]: I found all of the formats in here: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/datatypes_conf.xml.sample (was redirected there by the guide about [the format attribute](https://docs.galaxyproject.org/en/latest/dev/schema.html#id87)).
So I think that means they exist in Galaxy. Correct me if I'm wrong!

Comment (nicodr97) [reply tools/trimal/trimal.xml:73]: I looked here https://usegalaxy.org/datatypes but as your sources refer directly to tools I would say you are right. Thanks!

File: tools/trimal/trimal.xml [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    <command detect_errors="exit_code"><![CDATA[    
+        trimal -in '$in' -out '$trimmed_output' -htmlout '$html_summary' ${out_format_selector}
+        #if $trimming_mode.mode_selector == "custom"
+            -gapthreshold $trimming_mode.gapthreshold
+            -simthreshold $trimming_mode.simthreshold
+            -cons $trimming_mode.cons
+        #else:
+            $trimming_mode.mode_selector
+        #end if
+        ;
```
Comment (bgruening) [inline tools/trimal/trimal.xml:19]:
```suggestion
```

File: tools/trimal/trimal.xml [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+            </param>
+            <when value="-nogaps" />
+            <when value="-noallgaps"/>
+            <when value="-gappyout"/>
+            <when value="-strict"/>
+            <when value="-strictplus"/>
+            <when value="-automated1"/>
+            <when value="custom">
+                <param argument="-gapthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="gap threshold" help="1 - (fraction of sequences with a gap allowed). Range: [0 - 1]" multiple="false"/>
+                <param argument="-simthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="similarity threshold" help="Minimum average similarity allowed. Range: [0 - 1]" multiple="false"/>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:42]:
```suggestion
                <param argument="-simthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="Similarity threshold" help="Minimum average similarity allowed."/>
```

File: tools/trimal/trimal.xml [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+                <option value="custom">custom mode - eliminates a specified set of columns defined by the user.</option>
+            </param>
+            <when value="-nogaps" />
+            <when value="-noallgaps"/>
+            <when value="-gappyout"/>
+            <when value="-strict"/>
+            <when value="-strictplus"/>
+            <when value="-automated1"/>
+            <when value="custom">
+                <param argument="-gapthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="gap threshold" help="1 - (fraction of sequences with a gap allowed). Range: [0 - 1]" multiple="false"/>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:41]:
```suggestion
                <param argument="-gapthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="Gap threshold" help="1 - (fraction of sequences with a gap allowed)."/>
```

File: tools/trimal/trimal.xml [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+                <param argument="-simthreshold" type="float" optional="true" value="0.9" min="0.0" max="1.0" label="similarity threshold" help="Minimum average similarity allowed. Range: [0 - 1]" multiple="false"/>
+                <param argument="-cons" type="integer" optional="true" value="50" min="0" max="100" label="minimum conservance percentage" help="Minimum percentage of the positions in the original alignment to conserve." multiple="false"/>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:43]:
```suggestion
                <param argument="-cons" type="integer" optional="true" value="50" min="0" max="100" label="Minimum conservance percentage" help="Minimum percentage of the positions in the original alignment to conserve."/>
```

File: tools/trimal/trimal.xml [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+            <output name="html_summary" file="trimmed_example.009.AA.html"/>
+        </test>
+        <test expect_num_outputs="2">
+            <param name="in" value="example.009.AA.fasta"/>
+            <param name="mode_selector" value="custom" />
+            <param name="gapthreshold" value="0.5" />
+            <param name="simthreshold" value="0.5" />
+            <param name="cons" value="5" />
+            <param name="out_format_selector" value="-phylip_paml_m10" />
+            <output name="trimmed_output" file="custom_trimmed_example.009.AA.phy" />
```
Comment (bgruening) [inline tools/trimal/trimal.xml:95]:
```suggestion
            <output name="trimmed_output" file="custom_trimmed_example.009.AA.phy" ftype="phylip"/>
```

**Review by bgruening — COMMENTED** [review]
Very nice, last comments ...

File: tools/trimal/trimal.xml [hunk 15]
```diff
@@ -0,0 +1,128 @@
+<tool id="trimal" name="trimAl" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
+    <description>for automated alignment trimming</description>
+    <macros>
+        <token name="@TOOL_VERSION@">1.4.1</token>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:4]: 1.5 is already out ... do you want to use the latest version. This version is already 9 years old.

Comment (B0r1sD) [reply tools/trimal/trimal.xml:4]: Well I was not sure about this, see my last paragraph in the description in this pull request:

> Would also like to point out that Trimal 1.5.0 exists, but think we should get 1.4.1 wrapper properly finished and then release a new updated wrapper for the 1.5.0 version.

So best to do 1.5 immediately? The wrapper might need small rework potentially (see [deltas here](https://github.com/inab/trimal/compare/v1.4.1...v1.5.0)).

Comment (nicodr97) [reply tools/trimal/trimal.xml:4]: Just consider that most of that changes in files are related to tests and documentation. The part that affects the programme are only some additional features and fixed bugs.

Comment (bgruening) [reply tools/trimal/trimal.xml:4]: Both works for me. I would slightly prefer only the newer version.

Comment (B0r1sD) [reply tools/trimal/trimal.xml]: I simply bumped it to v1.5 and changed version line in test files (manually). The same tests still work so I guess this version bump went fine. The PR is now up for merging!

File: tools/trimal/trimal.xml [hunk 16]
```diff
@@ -0,0 +1,128 @@
+<tool id="trimal" name="trimAl" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
+    <description>for automated alignment trimming</description>
+    <macros>
+        <token name="@TOOL_VERSION@">1.5</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:6]: @B0r1sD do you have a bio.tools ID for this tool that you can add?

Comment (B0r1sD) [reply tools/trimal/trimal.xml:6]: Yes there is! I also added the same EDAM topics that I could find there.

**Comment by B0r1sD** [comment]
Small note: left in the test data file `example.004.AA.fasta` which is not in use but as it's so small (1 KB), could be useful for future wrapper tests.

File: tools/trimal/trimal.xml [hunk 17]
```diff
@@ -4,6 +4,14 @@
         <token name="@TOOL_VERSION@">1.5</token>
         <token name="@VERSION_SUFFIX@">0</token>
     </macros>
+    <edam_topics>
+        <edam_topic>topic_0080</edam_topic>
+        <edam_topic>topic_0160</edam_topic>
+        <edam_topic>topic_3168</edam_topic>
+    </edam_topics>
```
Comment (bgruening) [inline tools/trimal/trimal.xml:11]:
```suggestion
```

if we have bio.tools we get the edam for free from upstream :)

**Comment by bgruening** [comment]
Thanks a lot @B0r1sD!

**Pull request merged**